### DISCUSSION TITLE
(Noob) Help with Rule to publish wattage (when changed) using MQTT

### DISCUSSION THREAD

**Discussion by bhengeveld** [arendst/Tasmota#17620] [Q&A] [answered]
Dear Tasmota users

I'm using the powermeter function of Tasmota smartplugs for automations using Home Assistant. For instance; if I switch on my turntable, Home Assistant switches my receiver to the phono input. The trigger for Home Assistant is the published powerusage being above a certain value.

Now this works, but I'd like it to be as snappy as possible and with the minimum TelePeriod being 10 seconds, I'd think I'd be better off using Tasmota rules, but  that's where I'm stuck. (also I'd like the plug not to publish every 10 seconds) 

I'm getting to:
```Rule1 ON Power>5 DO publish tele/AT4/SENSOR endon```

but that's not working....

Below is a copy paste of the console of one of my plugs
```
16:39:26.416 MQT: tele/AT4/STATE = {"Time":"2023-01-06T16:39:26","Uptime":"0T23:29:40","UptimeSec":84580,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":5,"POWER":"ON","Wifi":{"AP":1,"SSId":"XXXXX","BSSId":"XX:XX:XX:XX:XX:XX","Channel":11,"Mode":"11n","RSSI":100,"Signal":-47,"LinkCount":3,"Downtime":"0T00:43:36"}}
16:39:26.429 MQT: tele/AT4/SENSOR = {"Time":"2023-01-06T16:39:26","ENERGY":{"TotalStartTime":"2022-12-20T21:52:49","Total":0.041,"Yesterday":0.018,"Today":0.023,"Period":0,"Power":12,"ApparentPower":23,"ReactivePower":20,"Factor":0.50,"Voltage":238,"Current":0.098}}
```
Hope you people can help me!

**Comment by sfromis** [comment, marked as answer]
If you want faster reporting, do not use rules, but instead `PowerDelta` to trigger reporting when consumption levels change. Beware that it may still not be very snappy, if you need quick reactions.

And if you want to use rules, start by reading about how they work, instead of doing something way off target.

**Reply by bhengeveld** [reply]
PowerDelta seems snappy enough, thank you!
